src/sqswatcher/plugins/slurm.py

```diff
@@ -100,16 +171,32 @@ def update_cluster(max_cluster_size, cluster_user, update_events, instance_prope
             node_name = "NodeName={0}".format(event.host.hostname)
             node_list = [node for node in node_list if node.split()[0] != node_name]
         elif event.action == "ADD":
-            # Add new node
-            new_node = "NodeName={nodename} CPUs={cpus} State=UNKNOWN\n".format(
-                nodename=event.host.hostname, cpus=event.host.slots
-            )
+            # Add new node and gres_node depending on whether the instance has GPU
+            if "gpus" in instance_properties:
+                new_node = "NodeName={nodename} CPUs={cpus} Gres=gpu:tesla:{gpus} State=UNKNOWN\n".format(
+                    nodename=event.host.hostname, cpus=event.host.slots, gpus=instance_properties["gpus"]
+                )
+            else:
+                new_node = "NodeName={nodename} CPUs={cpus} State=UNKNOWN\n".format(
+                    nodename=event.host.hostname, cpus=event.host.slots
+                )
+
             if new_node not in node_list:
                 node_list.append(new_node)
             # Restarting also if already in config cause it might have failed at the previous iteration
             nodes_to_restart.append(event.host.hostname)
 
+    return node_list, nodes_to_restart
+
+
+def update_cluster(max_cluster_size, cluster_user, update_events, instance_properties):
+    """Update and write node lists( and gres_nodes if instance has GPU); restart relevant nodes."""
+    node_list, nodes_to_restart = update_node_lists(update_events, instance_properties)
     _write_node_list(node_list, max_cluster_size, instance_properties)
+    # Only modify gres.conf if instance has GPU
+    if "gpus" in instance_properties:
```

Comment:
what if customer updates the cluster by changing instance type and goes from a gpu enabled instance to an instance without gpus? We should support this and clean up the gres.conf as nodes are deallocated. Always writing the gres.conf file might be a potential solution